static/js/protein/chart.ts

```diff
@@ -102,46 +130,119 @@ const TISSUE_COLOR_DICT = {
+// number to decide the ticks to be displayed
+const NUM_TICKS = 10;
+// graph specific dimensions
+const GRAPH_HEIGHT_1 = 200;
+const GRAPH_WIDTH_1 = 860;
+const GRAPH_HEIGHT_2 = 400;
+const GRAPH_WIDTH_2 = 700;
+const GRAPH_HEIGHT_3 = 500;
+const GRAPH_WIDTH_3 = 660;
+const GRAPH_WIDTH_4 = 760;
+// error point position
+const ERROR_POINT_POSITION_X1 = 450;
+const ERROR_POINT_POSITION_X2 = 470;
+const ERROR_POINT_POSITION_Y1 = 10;
+const ERROR_POINT_POSITION_Y2 = 30;
+const ERROR_POINT_POSITION_Y3 = 50;
 
-// interface for protein page datatypes which return string values
-export interface ProteinStrData {
-  name: string;
-  value: string;
+/**
+ * Gets the left and top coordinates of a rect element and positions the tooltip accordingly
+ * @param left_position
+ * @param top_position
+ */
+function mousemove(leftPosition: number, topPosition: number) {
+  TOOL_TIP.style("left", d3.event.offsetX + leftPosition + "px").style(
+    "top",
+    d3.event.offsetY + topPosition + "px"
+  );
+}
+/**
+ * Sets the opacity of the tooltip as zero, to hide it when the user hovers over other elements
+ */
+function mouseout() {
+  TOOL_TIP.style("opacity", 0);
+}
+/**
+ * Adds the x label to a graph based on user's input of width and height for label position, labelText for what the label reads, and svg for selecting the chart where the label is added
+ * @param width
+ * @param height
+ * @param labelText
+ * @param svg
+ */
+function addXLabel(
+  width: number,
+  height: number,
+  labelText: string,
+  svg: d3.Selection<SVGGElement, unknown, HTMLElement, any>
+) {
+  svg
+    .attr("class", "axis-label")
+    .append("text")
+    .attr(
+      "transform",
+      "translate(" + width / 2 + " ," + (height + MARGIN.top + 50) + ")"
+    )
+    .text(labelText);
 }
 
-// interface for variant gene associations for plotting error bars
-export interface VarGeneDataPoint {
-  id: string;
-  name: string;
-  value: number;
-  lower: number;
-  upper: number;
+/**
+ * Adds the y label to a graph based on user's input of width and height for label position, labelText for what the label reads, and svg for selecting the chart where the label is added
+ * @param width
+ * @param height
+ * @param labelText
+ * @param svg
+ */
+function addYLabel(
+  height: number,
+  labelText: string,
+  svg: d3.Selection<SVGGElement, unknown, HTMLElement, any>
+) {
+  svg
+    .append("text")
+    .attr("transform", "rotate(-90)")
+    .attr("y", 0 - MARGIN.left)
+    .attr("x", 0 - height / 2)
+    .attr("dy", "1em")
+    .text(labelText);
 }
 
 /**
- * Draw bar chart for tissue score.
+ * Draws the bar chart for protein-tissue association
+ * @param id
+ * @param data
+ * @returns
  */
 export function drawTissueScoreChart(id: string, data: ProteinStrData[]): void {
   // checks if the data is empty or not
   if (_.isEmpty(data)) {
     return;
   }
-  // reformats data to convert tissue score from string to number
   let reformattedData = [] as ProteinNumData[];
+  /**
```

Comment:
nit: sorry, I should've been more clear, this comment format is just for global functions, not the functions within other functions

Reply:
Got it!